tests/baselines/reference/inferConditionalConstraintMappedMember.types

```diff
@@ -9,7 +9,7 @@ type KeysWithoutStringIndex<T> =
 
 // Only "foo" | "bar" as expected, [string] index signature removed
 type test = KeysWithoutStringIndex<{ [index: string]: string; foo: string; bar: 'baz' }>
->test : never
+>test : "foo" | "bar"

```

Comment:
Fun fact - this test already showed the issue, we just weren't looking for it in this test. (Instead we were looking for a constraint error on a different line.)